### ISSUE TITLE
[feature request] Support output size parameter in Upsample ONNX operator

### ISSUE THREAD

**Issue by aachigorin** [pytorch/pytorch#7732]
Right now [Upsample](https://github.com/onnx/onnx/blob/master/docs/Operators.md#Upsample) in onnx supports fixed (calculated for specific input size) scale factors. Which are [calculated](https://github.com/pytorch/pytorch/blob/0a11018db690b02a52dd25b2470fdde794a8fac6/torch/onnx/symbolic.py#L498) during graph tracing based on current input and output sizes.

At the same time [F.upsample](https://pytorch.org/docs/master/nn.html?highlight=pad#torch.nn.functional.upsample) allows to specify exact output size, which is very convenient if we want to evaluate models with backward connections (like [Feature Pyramid Network](https://arxiv.org/pdf/1612.03144.pdf)) on inputs with varying sizes.

Could we extend onnx Upsample operator to support desired output size (exactly like in F.upsample), taken as additional input to the operator?

Simple example to check (this should export the graph with upsample taking its output size dynamically as h1.size()[-2:]):
```
class SimpleNet(nn.Module):
    def __init__(self):
        super(SimpleNet, self).__init__()
        self.conv1 = nn.Conv2d(3, 8, kernel_size=3, stride=1, padding=1)
        self.conv2 = nn.Conv2d(3, 8, kernel_size=3, stride=2, padding=1)
        
    def forward(self, x):
        h1 = self.conv1(x)
        h2 = F.upsample(x, size=h1.size()[-2:], mode='bilinear', align_corners=False)        
        h = torch.cat([h1, h2], dim=1)
        return h


model = SimpleNet()
save_path = '/Users/aachigorin/temp/test_bilinear_onnx_export.txt'
x = Variable(torch.randn(1, 3, 10, 10)).cpu()
torch_out = torch.onnx._export(model, x, save_path, export_params=True, verbose=True)
print('finished onnx export')

model = onnx.load(save_path)
print('check = {}'.format(onnx.checker.check_model(model)))
print(onnx.helper.printable_graph(model.graph))
```

cc @BowenBao @neginraoof

**Comment by ahirner** [comment]
Dynamic upsampling is a PR on onnx side: https://github.com/onnx/onnx/pull/1467

**Comment by wadimkehl** [comment]
Any update on this, seeing that the dependent PR has been merged? The code from above still produces an error with the latest nightly conda package ` 1.0.0.dev20181130-py3.6_cuda9.0.176_cudnn7.4.1_0`

` raise RuntimeError("ONNX symbolic expected a constant value in the trace")`

**Comment by glenn-jocher** [comment]
I'm experiencing the same problem when exporting from PyTorch 1.0.0 to ONNX to CoreML. Any known solution?

**Comment by Kay-Tian** [comment]
I'm experiencing the same issue, and by the way, the nightly version of '1.2.0.dev20190725+cpu' raise another fault：
`
 /anaconda3/envs/torch-nightly/lib/python3.6/site-packages/torch/nn/modules/conv.py(342)conv2d_forward()
    341         return F.conv2d(input, weight, self.bias, self.stride,
--> 342                         self.padding, self.dilation, self.groups)
    343 
Illegal instruction (core dumped)
`

**Comment by soumith** [comment]
@Kay-Tian this will be fixed in tomorrow's nightlies

**Comment by garymm** [comment]
`F.upsample` has been deprecated in favor of `torch.nn.functional.interpolate`, and the Upsample ONNX op has been deprecated since ONNX op set 10. The exporter now emits `Resize` instead.
Does this code produce the output you are asking for with the latest nightly pytorch? If not, could you please explain what you expect to see vs what you see?

```py
import torch
from torch import nn

class SimpleNet(nn.Module):                                                                                                                    
    def __init__(self):                                                                                                                        
        super(SimpleNet, self).__init__()                     
        self.conv1 = nn.Conv2d(3, 8, kernel_size=3, stride=1, padding=1)                                                                       
                                                                                                                                                                                                                                                                                                        
    def forward(self, x):                                      
        h1 = self.conv1(x)                                                                                                                     
        h2 = nn.functional.interpolate(x, size=h1.size()[-2:], mode='bilinear', align_corners=False)                                              
        h = torch.cat([h1, h2], dim=1)                                                                                                         
        return h                                                
                                                                       
                                                                                                                                                                                                                                                                                                           
model = SimpleNet()                                                                                                                                                                                                                                                                                    
x = torch.randn(1, 3, 10, 10).cpu()                                                                                                            
torch.onnx.export(model, x, "/dev/null", verbose=True, opset_version=13)
```

If it's not correct, what if you replace the last line with?

`torch.onnx.export(torch.jit.script(model), x, "/dev/null", example_outputs=model(x), verbose=True, opset_version=13)`

**Comment by garymm** [comment]
Closing due to lack of replies. If this is still an issue please open a new issue.
